async_consumer.py

```diff
@@ -0,0 +1,349 @@
+import pika
+import logging
+from structlog import wrap_logger
+import settings
+import time
+
+LOGGER = wrap_logger(
+    logging.getLogger(__name__)
+)
+
+
+class AsyncConsumer(object):
+    """This is an example consumer that will handle unexpected interactions
+    with RabbitMQ such as channel and connection closures.
+
+    If RabbitMQ closes the connection, it will reopen it. You should
+    look at the output, as there are limited reasons why the connection may
+    be closed, which usually are tied to permission related issues or
+    socket timeouts.
+
+    If the channel is closed, it will indicate a problem with one of the
+    commands that were issued and that should surface in the output as well.
+
+    """
+    EXCHANGE = 'message'
+    EXCHANGE_TYPE = 'topic'
+    QUEUE = settings.RABBIT_QUEUE
+
+    def __init__(self):
+        """Create a new instance of the consumer class, passing in the AMQP
+        URL used to connect to RabbitMQ.
+
+        :param str amqp_url: The AMQP url to connect with
+
+        """
+        self._connection = None
+        self._channel = None
+        self._closing = False
+        self._consumer_tag = None
+        self._url = None
+        self._stopped = False
+        self._amqpUrls = [
+            settings.RABBIT_URL,
+            settings.RABBIT_URL2,
+            settings.RABBIT_URL3
+        ]
+
+    def connect(self):
+        """This method connects to RabbitMQ, returning the connection handle.
+        When the connection is established, the on_connection_open method
+        will be invoked by pika.
+
+        :rtype: pika.SelectConnection
+
+        """
+        count = 1
+        while not self._stopped:
+            no_of_servers = len(self._amqpUrls)
+            server_choice = (count % no_of_servers) - 1
+            self._url = self._amqpUrls[server_choice]
+
+            try:
+                LOGGER.info('Connecting', amqp_url=self._url)
```

Comment:
I think it would be useful to add the count to this log line so we know how many times a reconnection has been attempted